test/abstract_syntax_tree.coffee

```diff
@@ -1245,6 +1245,36 @@ test "AST as expected for Obj node", ->
     ]
     implicit: yes
 
+  testExpression '''
+    a:
+      if b then c
+  ''',
+    type: 'ObjectExpression'
+    properties: [
+      type: 'ObjectProperty'
+      key: ID 'a'
+      value:
+        type: 'ConditionalExpression'
```

Comment:
This tests the specific case I ran into working on the Prettier plugin - this was generating an `IfStatement` inside a block instead of a `ConditionalExpression`